### PR TITLE
[IMP] : dispaly warning message on cart ,if product is out of stock and ...

### PR DESCRIPTION
**Pull request by vidhin6**
...not of type service or consumable
TASK: https://www.odoo.com/web#id=11457&view_type=form&model=project.task&action=333&active_id=131
Pad: https://pad.odoo.com/p/r.hjHDaFRnr7QNH3LH

### CODE REVIEW

File: addons/website_sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, set_qty=0, context=None, **kwargs):
+        """
+            display a out of stock warning message on cart, if customize option `Out of Stock Warning`
+            is set active and product type is not `Consumable` or `Service`
+         """
+        values = {}
+        customize_option = self.pool['ir.model.data'].xmlid_to_object(cr, SUPERUSER_ID, 'website_sale_stock.products_out_of_stock_warning')
+        line = self.pool['sale.order.line'].browse(cr, uid, line_id, context)
+        if customize_option.active and line and line.product_id.type not in ('consu', 'service') and (set_qty > int(line.product_id.virtual_available)):
```
Comment (vme-odoo) [inline addons/website_sale_stock/models/sale_order.py]: what if when line_id is None

File: addons/website_sale_stock/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            values['warning'] = _('Were sorry! We are able to accommodate only %s units of %s') % (int(line.product_id.virtual_available), line.product_id.name_get()[0][1])
```
Comment (vme-odoo) [inline addons/website_sale_stock/models/sale_order.py]: message not seems correct.

File: addons/website_sale_stock/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class sale_order(osv.Model):
+    _inherit = "sale.order"
+
+    def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, set_qty=0, context=None, **kwargs):
+        """
+            display a out of stock warning message on cart, if customize option `Out of Stock Warning`
+            is set active and product type is not `Consumable` or `Service`
+         """
+        values = {}
+        customize_option = self.pool['ir.model.data'].xmlid_to_object(cr, SUPERUSER_ID, 'website_sale_stock.products_out_of_stock_warning')
```
Comment (vme-odoo) [inline addons/website_sale_stock/models/sale_order.py]: can we get this customize_option from front site?

Comment (aja-odoo) [inline addons/website_sale_stock/models/sale_order.py]: yes , to check customize option is active or not , if active we will show a warning message.

File: addons/website_sale_stock/static/src/js/website_sale_stock.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+$(document).ready(function () {
+    //show warning outofstock, when click on add to cart from shop ,trigger on change of quantity input
+    if(window.location.pathname === '/shop/cart' && $('.oe_cart .show_outofstock_warning').length) {
+        $('.oe_cart input.js_quantity').trigger('change');
+    }
+});
```
Comment (vme-odoo) [inline addons/website_sale_stock/static/src/js/website_sale_stock.js]: Looks like work around solution, please try to fix other way.

Comment (aja-odoo) [inline addons/website_sale_stock/static/src/js/website_sale_stock.js]: @vme-odoo  : when we click on "Add to Cart"  button from product/general shop view , we are not able to trigger "js_quantity" (quantity input box), so when we click on "Add to Cart" and redirect to "/shop/cart" , we need to forcefully trigger a quantity input box to show alert , i think this is better way to do it.

Comment (vme-odoo) [inline addons/website_sale_stock/static/src/js/website_sale_stock.js]: if "/shop/cart" change to something else than what?

Comment (aja-odoo) [inline addons/website_sale_stock/static/src/js/website_sale_stock.js]: ok , then we can remove `window.location.pathname === '/shop/cart'` and relay on `$('.oe_cart .show_outofstock_warning').length)` , when we got class `.show_outofstock_warning` we trigger input , is it ok ?

**Comment by rim-odoo** [comment]
https://github.com/odoo/odoo/commit/13dae8847ea0b034283c11836d7be20fda49ff37

**Pull request closed without merging**